campaign-launcher/client/src/components/CampaignAddress/index.tsx

```diff
@@ -0,0 +1,90 @@
+import { FC, MouseEvent, useState } from "react"
+
+import { ChainId } from "@human-protocol/sdk"
+import ContentCopyIcon from '@mui/icons-material/ContentCopy';
+import { Box, IconButton, Tooltip, Typography } from "@mui/material"
+
+import { useIsMobile } from "../../hooks/useBreakpoints";
+import { OpenInNewIcon } from "../../icons"
+import { formatAddress, getExplorerUrl } from "../../utils"
+
+type Props = {
+  address: string;
+  chainId: ChainId;
+}
+
+const handleAddressClick = (
+  e: MouseEvent<HTMLButtonElement>,
+  chainId: ChainId,
+  address: string
+) => {
+  e.stopPropagation();
+  const explorerUrl = getExplorerUrl(chainId, address);
+  window.open(explorerUrl, '_blank');
+};
+
+const CampaignAddress: FC<Props> = ({ address, chainId }) => {
+  const [isCopied, setIsCopied] = useState(false);
+
+  const isMobile = useIsMobile();
+
+  const handleCopy = (e: MouseEvent<HTMLButtonElement>) => {
+    e.stopPropagation();
+    navigator.clipboard.writeText(address);
+    setIsCopied(true);
+    setTimeout(() => {
+      setIsCopied(false);
+    }, 1500);
+  }
+
+  return (
+    <Box 
+      display="flex" 
+      alignItems="center" 
+      minWidth="250px"
+      maxWidth={{ xs: "100%", sm: "fit-content" }}
+      flexGrow={1}
+      flexShrink={1}
+      gap={1} 
+      py={2} 
+      px={3} 
+      borderRadius="40px" 
+      border="1px solid" 
+      borderColor="divider"
+    >
+      <Typography 
+        variant="subtitle2" 
+        textOverflow="ellipsis" 
+        overflow="hidden" 
+        whiteSpace="nowrap"
+      >
+        {isMobile ? formatAddress(address) : address}
+      </Typography>
+      <IconButton
```

Comment:
Maybe have it disabled if `isCopied = true`? Otherwise we can start many timers
Or maybe even different icon (e.g. `check`)